internal/cri/server/sandbox_run.go

```diff
@@ -358,6 +358,15 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		sandboxCreateNetworkTimer.UpdateSince(netStart)
 	}
 
+	if err := sandboxInfo.AddExtension(podsandbox.MetadataKey, &sandbox.Metadata); err != nil {
```

Comment:
The L323 has updated sandbox for extension. do we still need this line?

Reply:
It's before the IP gets assigned however (in setupPodNetwork). Perhaps we should just get rid of that metadata save. I'm hesitant to because of how fickle these saves seem to be..

Reply:
Maybe this IP information has been broken since https://github.com/containerd/containerd/pull/7679.

I think we should add e2e test to cover it.

Reply:
The L410 has added sandbox into store. The ip is from that store.

Reply:
>Maybe this IP information has been broken since https://github.com/containerd/containerd/pull/7679.
I think so yea. That's a good idea, let me see how the other userns integration tests function. I added a recovery test to verify we still have ip information before but for regular containers.

Reply:
>The L410 has added sandbox into store. The ip is from that store.

That's our in-memory store iirc. That's the reason before restart everything is fine, but after a restart that is not present anymore and we need to consult (and re-populate that in-memory store) with what's on disk.